File: src/main/java/core/procedures/strings/RePattern.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

import java.util.regex.Pattern;

public final class RePattern extends AFn {

  public RePattern() {
    super(new FnArgsBuilder().min(1).max(1).mandatory(new Class[]{CharSequence.class}).build());
  }

  @Override
  public String getName() {
    return "re-pattern";
  }

  @Override
  public Pattern apply1(Object arg) {
    return Pattern.compile(arg.toString());
  }
}

File: src/main/java/core/procedures/strings/RePattern.kt
package core.procedures.strings

import core.procedures.AFn
import core.procedures.Arity.Exactly

import java.util.regex.Pattern

class RePattern : AFn<CharSequence?, Pattern>(name = "re-pattern", isPure = true, arity = Exactly(1),
                                              mandatoryArgsTypes = arrayOf(CharSequence::class.java)) {

    override operator fun invoke(arg: CharSequence?): Pattern = Pattern.compile(arg!!.toString())
}
